visensor_calibration_flasher/src/visensor_calibration_flasher.cpp

```diff
@@ -112,63 +119,106 @@ int main(int argc, char **argv)
+      lens_model->setType();
+      lens_model->k1_ = (double) distortion_coeffs[0];
+      lens_model->k2_ = (double) distortion_coeffs[1];
+      lens_model->k3_ = (double) distortion_coeffs[2];
+      lens_model->k4_ = (double) distortion_coeffs[3];
+    }
+    else {
+      std::cout << "Distortion Model is not supported. Abort, abort!\n";
+      return 1;
+    }
 
-    camera_calibration.focal_point[0] = (double) intrinsics[0];
-    camera_calibration.focal_point[1] = (double) intrinsics[1];
-    camera_calibration.principal_point[0] = (double) intrinsics[2];
-    camera_calibration.principal_point[1] = (double) intrinsics[3];
+    std::cout << "lens model is: " << static_cast<int>(camera_calibration.lens_model_->type_) << std::endl;
+
+    camera_model.assign(cam_params["camera_model"]);
+    XmlRpc::XmlRpcValue intrinsics = cam_params["intrinsics"];
+    std::cout << "projection_model is: " << camera_model << std::endl;
+    if (camera_model == std::string("pinhole")) {
+      visensor::ViCameraProjectionModelPinhole::Ptr projection_model = camera_calibration.getProjectionModel<visensor::ViCameraProjectionModelPinhole>();
+      projection_model->setType();
+      projection_model->focal_length_u_ = (double) intrinsics[0];
+      projection_model->focal_length_v_ = (double) intrinsics[1];
+      projection_model->principal_point_u_ = (double) intrinsics[2];
+      projection_model->principal_point_v_ = (double) intrinsics[3];
+    }
+    else if (camera_model == std::string("omnidirectional")){
+      visensor::ViCameraProjectionModelOmnidirectional::Ptr projection_model = camera_calibration.getProjectionModel<visensor::ViCameraProjectionModelOmnidirectional>();
+      projection_model->setType();
+      projection_model->focal_length_u_ = (double) intrinsics[0];
+      projection_model->focal_length_v_ = (double) intrinsics[1];
+      projection_model->principal_point_u_ = (double) intrinsics[2];
+      projection_model->principal_point_v_ = (double) intrinsics[3];
+      // todo(lschmid) how to check if intrinsics has 5 attributes?
+      projection_model->mirror_xi_ = (double) intrinsics[4];
+    }
+    else {
+      std::cout << "Current Camera Model is supported. Abort, abort!\n";
+      return 1;
+    }
 
     XmlRpc::XmlRpcValue resolution = cam_params["resolution"];
+    std::cout << "resolution is: " << resolution << std::endl;
+    camera_calibration.resolution_[0] = (int) resolution[0];
+    camera_calibration.resolution_[1] = (int) resolution[1];
 
-    std::string camera_model;
-    std::string distortion_model;
 
-    camera_model.assign(cam_params["camera_model"]);
-    distortion_model.assign(cam_params["distortion_model"]);
+    camera_calibration.slot_id_ = 0;
+
+
+    // delete every factroy calibration of the corresponding cam
+    try {
+      privat_drv->cleanCameraCalibrations(camera_id, 0, -1,
```

Comment:
As mentioned in libvisensor pull request -> Can we have a function such as cleanCameraCalibrations(camera_id) which wraps this function call?